test/services/payout-service-test.ts

```diff
@@ -0,0 +1,261 @@
+            rewardAmount.mul(3)
+        );
+
+        const expectedAssetSimplePayoutId = 0;
+        await testData.payoutManager.connect(testData.issuerOwner).createPayout({
+            asset: assetSimple.address,
+            totalAssetAmount: payoutInfo.totalAssetAmount,
+            ignoredAssetAddresses: payoutInfo.ignoredAssetAddresses,
+            payoutInfo: payoutInfo.payoutInfo,
+            assetSnapshotMerkleRoot: payoutInfo.assetSnapshotMerkleRoot,
+            assetSnapshotMerkleDepth: payoutInfo.assetSnapshotMerkleDepth,
+            assetSnapshotBlockNumber: payoutInfo.assetSnapshotBlockNumber,
+            assetSnapshotMerkleIpfsHash: payoutInfo.assetSnapshotMerkleIpfsHash,
+            rewardAsset: payoutInfo.rewardAsset,
+            totalRewardAmount: payoutInfo.totalRewardAmount
+        });
+
+        const expectedAssetTransferablePayoutId = 1;
+        await testData.payoutManager.connect(testData.issuerOwner).createPayout({
+            asset: assetTransferable.address,
+            totalAssetAmount: payoutInfo.totalAssetAmount,
+            ignoredAssetAddresses: payoutInfo.ignoredAssetAddresses,
+            payoutInfo: payoutInfo.payoutInfo,
+            assetSnapshotMerkleRoot: payoutInfo.assetSnapshotMerkleRoot,
+            assetSnapshotMerkleDepth: payoutInfo.assetSnapshotMerkleDepth,
+            assetSnapshotBlockNumber: payoutInfo.assetSnapshotBlockNumber,
+            assetSnapshotMerkleIpfsHash: payoutInfo.assetSnapshotMerkleIpfsHash,
+            rewardAsset: payoutInfo.rewardAsset,
+            totalRewardAmount: payoutInfo.totalRewardAmount
+        });
+
+        const expectedAssetBasicPayoutId = 2;
+        await testData.payoutManager.connect(testData.issuerOwner).createPayout({
+            asset: assetBasic.address,
+            totalAssetAmount: payoutInfo.totalAssetAmount,
+            ignoredAssetAddresses: payoutInfo.ignoredAssetAddresses,
+            payoutInfo: payoutInfo.payoutInfo,
+            assetSnapshotMerkleRoot: payoutInfo.assetSnapshotMerkleRoot,
+            assetSnapshotMerkleDepth: payoutInfo.assetSnapshotMerkleDepth,
+            assetSnapshotBlockNumber: payoutInfo.assetSnapshotBlockNumber,
+            assetSnapshotMerkleIpfsHash: payoutInfo.assetSnapshotMerkleIpfsHash,
+            rewardAsset: payoutInfo.rewardAsset,
+            totalRewardAmount: payoutInfo.totalRewardAmount
+        });
+
+        const payoutsAfterCreation = await testData.payoutService.getPayoutsForIssuer(
+            testData.issuer.address,
+            testData.payoutManager.address,
+            [
+                testData.assetFactory.address,
+                testData.assetTransferableFactory.address,
+                testData.assetSimpleFactory.address
+            ]
+        );
+
+        expect(payoutsAfterCreation.length).to.be.equal(3);
+        
+        const assetBasicPayout: PayoutForIssuerItem = 
+            payoutsAfterCreation.filter(item => item.asset == assetBasic.address)[0];
```

Comment:
Just info, you could use find for this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find